pkg/trie/triedb/codec/node.go

```diff
@@ -23,11 +24,11 @@ type (
 	// InlineNode contains bytes of the encoded node data
 	InlineNode []byte
 	// HashedNode contains a hash used to lookup in db for encoded node data
-	HashedNode common.Hash
+	HashedNode[H any] struct{ Hash H }
```

Comment:
Why not?

```suggestion
	HashedNode[H hash.Hash] struct{ Hash H }
```

Reply:
I lean towards trying to be succinct and generic as possible per type.  I think it's understood that `hash.Hash` is used exclusively in `TrieDB`, but it doesn't necessarily mean that we need it to be for `HashedNode`.  If you feel strongly the other way I'm open to changing it.

Reply:
mmm since we are always storing hashes in `HashedNode` I would say that it will be right to use the `hash.Hash` instead of `any` to prevent issues and also being consistent with `HashedValue[H hash.Hash]`. What do you think? do you see any issue?

Reply:
I addressed this in #4239 see commit: bea5ad3448ecf2ff84763082f5db3246c216726c